indico/queries/submission.py

```diff
@@ -0,0 +1,194 @@
+
+    Args:
+        submission_ids (List[int], optional): Submission ids to filter by
+        workflow_ids (List[int], optional): Workflow ids to filter by
+        filters (SubmissionFilter or Dict, optional): Submission attributes to filter by
+        limit (int, optional): Maximum number of Submissions to return. Defaults to 1000
+
+    Returns:
+        List[Submission]: All the found Submission objects
+    """
+
+    query = """
+        query ListSubmissions(
+            $submissionIds: [Int],
+            $workflowIds: [Int],
+            $filters: SubmissionFilter,
+            $limit: Int
+        ){
+            submissions(
+                submissionIds: $submissionIds,
+                workflowIds: $workflowIds,
+                filters: $filters,
+                limit: $limit
+            ){
+                submissions {
+                    id
+                    datasetId
+                    workflowId
+                    status
+                    inputFile
+                    inputFilename
+                    resultFile
+                }
+            }
+        }
+    """
+
+    def __init__(
+        self,
+        submission_ids: List[int] = None,
+        workflow_ids: List[int] = None,
+        filters: Union[Dict, SubmissionFilter] = None,
+        limit: int = 1000,
+    ):
+        super().__init__(
+            self.query,
+            variables={
+                "submissionIds": submission_ids,
+                "workflowIds": workflow_ids,
+                "filters": filters,
+                "limit": limit,
+            },
+        )
+
+    def process_response(self, response) -> List[Submission]:
+        return [
+            Submission(**s)
+            for s in super().process_response(response)["submissions"]["submissions"]
+        ]
+
+
+class GetSubmission(GraphQLRequest):
+    """
+    Retrieve a Submission by id
+
+    Args:
+        submission_id (int): Submission id
+
+    Returns:
+        Submission: Found Submission object
+    """
+
+    query = """
+        query GetSubmission($submissionId: Int!){
+            submission(id: $submissionId){
+                id
+                datasetId
+                workflowId
+                status
+                inputFile
+                inputFilename
+                resultFile
+            }
+        }
+    """
+
+    def __init__(
+        self, submission_id: int,
+    ):
+        super().__init__(
+            self.query, variables={"submissionId": submission_id},
+        )
+
+    def process_response(self, response) -> Submission:
+        return Submission(**(super().process_response(response)["submission"]))
+
+
+class GenerateSubmissionResult(GraphQLRequest):
+    query = """
+        mutation CreateSubmissionResults($submissionId: Int!) {
+            submissionResults(submissionId: $submissionId) {
+                jobId
+            }
+        }
+
+    """
+
+    def __init__(
+        self, submission_id: int
```

Comment:
you could also probably take a submission object here as an alternative to make it even easier. 

submission: Union[int, Submission] or just as two different params you check submission_id: int =None, submission: Submission